### PR TITLE
[Enhance] Refactor the loss function in HTC and SCNet

### PR DESCRIPTION
**Pull request by AronLin**
Now we support `ignore_index` in `CrossEntropyLoss`.

We should refactor the loss function in HTC and SCNet to use our own CrossEntropyLoss instead of PyTorch.

### CODE REVIEW

File: mmdet/models/roi_heads/mask_heads/fused_semantic_head.py [hunk 1]
```diff
@@ -30,10 +30,12 @@ def __init__(self,
                  in_channels=256,
                  conv_out_channels=256,
                  num_classes=183,
-                 ignore_label=255,
```
Comment (ZwwWayne) [inline mmdet/models/roi_heads/mask_heads/fused_semantic_head.py:33]: should not delete them immediately. We should keep them and add deprecated warning for this two argument for three month at least

**Review by hhaAndroid — COMMENTED** [review]
OK

**Comment by AronLin** [comment]
The training results for htc_r50_fpn_1x_coco:

Backbone  |  box AP | mask AP | 
|:---------:|:-------:|:-------:|
| origin |     42.3 |  37.4 |
| now   |     42.1 |  37.5 |

**Review by ZwwWayne — APPROVED** [review]
